### PR TITLE
chore(install): NR CLI Translate base64 encoded urls command #623

### PR DESCRIPTION
**Pull request by GlenOFoghlu**
Created this PR to create Decode CLI operation -> https://github.com/newrelic/newrelic-cli/issues/623

### CODE REVIEW

File: cmd/newrelic/command_decode.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		relicString = u.Query().Get("pane")
+		decoded, err := base64.StdEncoding.DecodeString(relicString)
+
+		if err != nil {
+			fmt.Println("decode error:", err)
+		}
+
+		decodedURL := string(decoded)
+		fmt.Println("\n**** Decoded URL **** ")
+		fmt.Println(decodedURL)
```
Comment (sanderblue) [inline cmd/newrelic/command_decode.go:32]: Can we delete these two lines of output? This looks like debugging output 🙂

File: cmd/newrelic/command_decode.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+)
+
+// Command represents the decode command.
+var cmdDecode = &cobra.Command{
+	Use:     "decode",
+	Short:   "Decodes NR1 URL Strings ",
+	Example: `newrelic decode https://one.newrelic.com/launcher/nr1-core.home?pane=eyJuZXJkbGV0SWQiOiJucjEtY29yZS5ob21lLXNjcmVlbiJ9&platform[accountId]=1`,
+
+	Run: func(cmd *cobra.Command, args []string) {
+		relicString := strings.Join(args, " ")
```
Comment (sanderblue) [inline cmd/newrelic/command_decode.go:21]: Does the empty string need the space? 

i.e. can this
```go
strings.Join(args, " ")
```

be this?
```go
strings.Join(args, "")
```

File: cmd/newrelic/command_decode.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				}
+			}
+
+			decoded, err := base64.StdEncoding.DecodeString(decodedURL)
+			if err != nil {
+				fmt.Println("decode error:", err)
+			}
+
+			decodedURL := string(decoded)
+			fmt.Println("\n**** Decoded Entity ID **** ")
```
Comment (sanderblue) [inline cmd/newrelic/command_decode.go:53]: Awesome stuff to get this going. I think we can move forward with this, but we might want to adjust the command signature to be a little more declarative. A declarative signature will help us avoid needing the extra line of output to describe what we're displaying. 

Hypothetical command signature:
```zsh
newrelic decode url https://yadda.yadda.com?pane=eyJuZXJk... --param "pane" --search "entityId"
```

This signature allows the user to define what they want from the URL's query param `pane` as well as what to search for in the encoded string. The `--query-param` flag value can then be passed to the function on [line 23](https://github.com/newrelic/newrelic-cli/pull/653/files#diff-90f45227f18dce8728ec5764cfbef896385ae0d17ac97f9b37ff5d6b64bd2c18R23) and the `--search` value can be passed to [line 34](https://github.com/newrelic/newrelic-cli/pull/653/files#diff-90f45227f18dce8728ec5764cfbef896385ae0d17ac97f9b37ff5d6b64bd2c18R34) and [line 38](https://github.com/newrelic/newrelic-cli/pull/653/files#diff-90f45227f18dce8728ec5764cfbef896385ae0d17ac97f9b37ff5d6b64bd2c18R38) (`gjson.Get(decodedURL, decodeSearch)`) or something to that effect. 

This give the user the ability to decode other URL parts as well if/when things change down the line.

**Review by sanderblue — CHANGES_REQUESTED** [review]
This is looking great! I have one suggestion for the command signature now that I understand the desired output of the command.

**Comment by zlesnr** [comment]
Thank you for the contribution.

**Pull request closed without merging**